cmd/root.go

```diff
@@ -99,9 +141,83 @@ func postRunEOutput(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	if m := viper.GetString("module"); m != "" {
+		dm, err := mxwriter.NewDemux(hclOut)
+		if err != nil {
+			return err
+		}
+
+		moduleName := filepath.Base(m)
+		mdir := fmt.Sprintf("module-%s", moduleName)
+
+		err = os.Mkdir(filepath.Join(m, mdir), 0700)
+		if err != nil {
+			return err
+		}
+
+		for _, aux := range dm.Keys() {
+			k := aux
+			var (
+				filep string
+			)
+			if k == hcl.ModuleCategoryKey {
+				filep = filepath.Join(m, "module.tf")
+			} else {
+				filep = filepath.Join(m, mdir, fmt.Sprintf("%s.tf", k))
+			}
+
+			f, err := os.OpenFile(filep, os.O_APPEND|os.O_TRUNC|os.O_RDWR|os.O_CREATE, 0644)
+			if err != nil {
+				return fmt.Errorf("could not OpenFile %s because: %s", filep, err)
+			}
+			defer f.Close()
```

Comment:
That's a lot of ongoing defers. Shouldn't we close the files as we go?

Reply:
Yeh I thought the same, I had it but remove it afterwards, I'll re-add it normally, no defer.